### ISSUE TITLE
java.lang.RuntimeException for com.pushwoosh:pushwoosh:4.8.4

### ISSUE THREAD

**Issue by katepratik** [Pushwoosh/pushwoosh-android-sdk#48]
Hey!

I am getting blow crash on GI-I9500_TMMARS Device using Android 4.1.1 
Have you fixed this issue on latest SDK?

```
Fatal Exception: java.lang.RuntimeException: Unable to instantiate receiver com.pushwoosh.MessageAlertReceiver: java.lang.ClassNotFoundException: com.pushwoosh.MessageAlertReceiver
       at android.app.ActivityThread.handleReceiver(ActivityThread.java:2248)
       at android.app.ActivityThread.access$1500(ActivityThread.java:131)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1272)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:137)
       at android.app.ActivityThread.main(ActivityThread.java:4791)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:511)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
       at dalvik.system.NativeStart.main(NativeStart.java)
Caused by java.lang.ClassNotFoundException: com.pushwoosh.MessageAlertReceiver
       at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:61)
       at java.lang.ClassLoader.loadClass(ClassLoader.java:501)
       at java.lang.ClassLoader.loadClass(ClassLoader.java:461)
       at android.app.ActivityThread.handleReceiver(ActivityThread.java:2243)
       at android.app.ActivityThread.access$1500(ActivityThread.java:131)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1272)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:137)
       at android.app.ActivityThread.main(ActivityThread.java:4791)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:511)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
       at dalvik.system.NativeStart.main(NativeStart.java)
```

**Comment by shaders** [comment]
That's weird, this receiver is for Amazon only. It should be instantiated on Android at all.

Could you show me your AndroidManifest.xml?

**Comment by katepratik** [comment]
What exactly are you looking for? I will share that part of my Mainfest with you.

**Comment by shaders** [comment]
Something like this (or any other part with MessageAlertReceiver):
```
        <receiver
            android:name="com.google.android.gms.gcm.GcmReceiver"
            android:exported="true"
            android:permission="com.google.android.c2dm.permission.SEND" >
            <intent-filter>
                <action android:name="com.google.android.c2dm.intent.RECEIVE" />

                <category android:name="${applicationId}" />
            </intent-filter>
        </receiver>
        <receiver
            android:name="com.pushwoosh.MessageAlertReceiver"
            android:permission="com.amazon.device.messaging.permission.SEND" >
            <intent-filter>
                <action android:name="com.amazon.device.messaging.intent.REGISTRATION" />
                <action android:name="com.amazon.device.messaging.intent.RECEIVE" />

                <category android:name="${applicationId}" />
            </intent-filter>
        </receiver>
```

**Comment by katepratik** [comment]
I have this part
`<receiver
            android:name="com.mixpanel.android.mpmetrics.GCMReceiver"
            android:permission="com.google.android.c2dm.permission.SEND">
            <intent-filter>
                <action android:name="com.google.android.c2dm.intent.RECEIVE" />
                <action android:name="com.google.android.c2dm.intent.REGISTRATION" />

                <category android:name="${applicationId}" />
            </intent-filter>
        </receiver>`

I don't have this one which you specified above 
`        <receiver
            android:name="com.pushwoosh.MessageAlertReceiver"
            android:permission="com.amazon.device.messaging.permission.SEND" >
            <intent-filter>
                <action android:name="com.amazon.device.messaging.intent.REGISTRATION" />
                <action android:name="com.amazon.device.messaging.intent.RECEIVE" />

                <category android:name="${applicationId}" />
            </intent-filter>
        </receiver>`

**Comment by DimanAM** [comment]
`com.pushwoosh.MessageAlertReceiver` is defined in Pushwoosh SDK. However it is subclass of `com.amazon.device.messaging.ADMMessageReceiver` which is presented only on Amazon devices. This receiver can receive intents only on Amazon. It is not exported and is protected by permission.

Does this issue occur on real application or in test environment? How often does it happen?

As a workaround you can create stub for `com.amazon.device.messaging.ADMMessageReceiver`:

```java
public class ADMMessageReceiver extends android.content.BroadcastReceiver {

    public ADMMessageReceiver(Class<?> clazz) {

    }

    @Override
    public void onReceive(Context context, Intent intent) {

    }
}
```

And `com.amazon.device.messaging.ADMMessageHandlerBase`:

```java
public class ADMMessageHandlerBase {
}
```

**Comment by DimanAM** [comment]
Should be fixed in [4.12.0](https://github.com/Pushwoosh/pushwoosh-android-sdk/releases/tag/4.12.0)
